pkg/kics/sink.go

```diff
@@ -20,6 +20,11 @@ func (s *Service) sink(ctx context.Context, filename, scanID string, rc io.Reade
 		return errors.Wrapf(err, "failed to get file content: %s", filename)
 	}
 
+	content, err = s.Parser.Resolve(*content, filename)
+	if err != nil {
+		return errors.Wrapf(err, "failed to resolve file content: %s", filename)
+	}
+
```

Comment:
Since each parser has its own resolver, can't we add this resolve function to pkg/parser/parser.go Parse?
ex: 
func (c *Parser) Parse(filePath string, fileContent []byte) ([]model.Document, model.FileKind, error) {
	ext := filepath.Ext(filePath)
	if ext == "" {
		ext = filepath.Base(filePath)
	}
	if p, ok := c.parsers[ext]; ok {
        resolved, err := p.Resolve(fileContent, filename)
		if err != nil {
			return nil, err
		}
		obj, err := p.Parse(filePath, resolved)
		if err != nil {
			return nil, "", err
		}

		return obj, p.GetKind(), nil
	}

	return nil, "", ErrNotSupportedFile
}